packages/cardano-services/test/StakePoolSeach/StakePoolSearchHttpServer.test.ts

```diff
@@ -0,0 +1,488 @@
+        });
+        it('search by retiring status', async () => {
+          const req: StakePoolQueryOptions = {
+            filters: {
+              status: [Cardano.StakePoolStatus.Retiring]
+            }
+          };
+          const responseWithOrCondition = await doServerRequest(setFilterCondition(req, 'or'));
+          const response = await doServerRequest(req);
+          expect(responseWithOrCondition).toMatchSnapshot();
+          expect(response).toEqual(responseWithOrCondition);
+        });
+      });
+      describe('search pools by pledge met', () => {
+        it('search by pledge met on true', async () => {
+          const req: StakePoolQueryOptions = {
+            filters: {
+              pledgeMet: true
+            }
+          };
+          const responseWithAndCondition = await doServerRequest(req);
+          const responseWithOrCondition = await got.post(`${apiUrlBase}/search`, {
+            json: { args: [setFilterCondition(req, 'or')] }
+          });
+          expect(responseWithOrCondition.statusCode).toEqual(200);
+          expect(JSON.parse(responseWithOrCondition.body)).toEqual(responseWithAndCondition);
+        });
+        // FIXME: throws 500 error when running after previous test
+        //        if running by itself or with previous test skipped doesn't throw and fails because of equality
+        it('search by pledge met on false', async () => {
+          const req = {
+            filters: {
+              pledgeMet: false
+            }
+          };
+          const responseWithOrCondition = await doServerRequest(setFilterCondition(req, 'or'));
+          expect(responseWithOrCondition).toMatchSnapshot();
+          const responseWithAndCondition = await doServerRequest(req);
+          expect(responseWithAndCondition).toEqual(responseWithAndCondition);
+        });
+      });
+      describe('search pools by multiple filters', () => {
+        const req: StakePoolQueryOptions = {
+          filters: {
+            identifier: {
+              _condition: 'or',
+              values: [
+                { name: 'THE AMSTERDAM NODE' },
+                { name: 'banderini' },
+                { ticker: 'TEST' },
+                { id: Cardano.PoolId('pool1jcwn98a6rqr7a7yakanm5sz6asx9gfjsr343mus0tsye23wmg70') }
+              ]
+            }
+          }
+        };
+        const reqWithMultipleFilters: StakePoolQueryOptions = {
+          filters: {
+            ...req.filters,
+            _condition: 'or',
+            status: [
+              Cardano.StakePoolStatus.Activating,
+              Cardano.StakePoolStatus.Active,
+              Cardano.StakePoolStatus.Retired,
+              Cardano.StakePoolStatus.Retiring
+            ]
+          }
+        };
+        // TODO: the status code is only being checked in the following tests in order to just validate query errors.
+        // As an improve, the stake pools response could be validated too.
```

Comment:
This comment is probably the same as the above `it.todo`?

Reply:
I think so, this is what you meant, right? I think I can include in my current task

Reply:
Yes, that would be great